L/llama_cpp/build_tarballs.jl

```diff
@@ -1,12 +1,18 @@
 using BinaryBuilder, Pkg
 
 name = "llama_cpp"
-version = v"0.0.6"  # fake version number
+version = v"0.0.7"  # fake version number
 
 # url = "https://github.com/ggerganov/llama.cpp"
 # description = "Port of Facebook's LLaMA model in C/C++"
 
 # NOTES
+# - _mm_loadu_si64 needs at least gcc-11.3.0
+#   https://stackoverflow.com/questions/72837929/mm-loadu-si32-not-recognized-by-gcc-on-ubuntu
+#   gcc-11.1.0 will silently produce wrong code, so we have to use gcc-12
+#   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=99754
+# - windows and mmap working?
+#   warning: You are building for pre-Windows 8; prefetch not supported
```

Comment:
Would defining [`_WIN32_WINNT`/`WINVER`](https://learn.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt?view=msvc-170) appropriately help here since Julia [requires Windows 10](https://julialang.org/downloads/#supported_platforms) anyway?  Maybe we should just set those macros for Windows 10 in BB directly, PRs welcome.

Reply:
I will try a patch here for llama_cpp.

Where would the patch for BB go?

Reply:
> I will try a patch here for llama_cpp.

What patch?  It's just a macro to pass to the compiler.

> Where would the patch for BB go?

https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/df99a14849abf2f97aef595de8265acea6877b1e/src/Runner.jl#L461-L482
https://github.com/JuliaPackaging/BinaryBuilderBase.jl/blob/df99a14849abf2f97aef595de8265acea6877b1e/src/Runner.jl#L316-L337

Reply:
PR for BB here: https://github.com/JuliaPackaging/BinaryBuilderBase.jl/pull/308